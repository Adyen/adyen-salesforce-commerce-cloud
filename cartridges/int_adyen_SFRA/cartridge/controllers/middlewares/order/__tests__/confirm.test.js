"use strict";

/* eslint-disable global-require */
var confirm;
var req;
var res;
beforeEach(function () {
  var _require = require('../../index'),
    order = _require.order;
  confirm = order.confirm;
  jest.clearAllMocks();
  res = {
    setViewData: jest.fn(),
    getViewData: jest.fn(function () {
      return {};
    })
  };
  req = {
    querystring: {
      ID: 'mocked_querystring_id',
      token: 'mocked_token'
    }
  };
});
afterEach(function () {
  jest.resetModules();
});
describe('Confirm', function () {
  it('should do nothing if giving is not enabled', function () {
    var AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
    AdyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
    AdyenHelper.getAdyenGivingConfig.mockImplementation(function () {
      return null;
    });
    confirm(req, res, jest.fn());
    expect(res.setViewData).toBeCalledTimes(0);
  });
  it('should do nothing if giving is not available', function () {
    var AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
    AdyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
    AdyenHelper.isAdyenGivingAvailable.mockImplementation(function () {
      return false;
    });
    confirm(req, res, jest.fn());
    expect(res.setViewData).toBeCalledTimes(0);
  });
  it('should set view data', function () {
    confirm(req, res, jest.fn());
    expect(res.setViewData).toMatchSnapshot();
  });
});