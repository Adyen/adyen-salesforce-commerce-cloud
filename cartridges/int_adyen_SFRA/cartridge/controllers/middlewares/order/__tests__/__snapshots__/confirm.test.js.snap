// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm should set view data 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "adyen": {
          "adyenGivingAvailable": true,
          "adyenGivingBackgroundUrl": "mocked_background_url",
          "adyenGivingLogoUrl": "mocked_logo_url",
          "charityDescription": "mocked_charity_description",
          "charityName": "mocked_charity_name",
          "charityWebsite": "mocked_charity_website",
          "clientKey": "mocked_client_key",
          "donationAmounts": "{"currency":"EUR","values":[10,20,30]}",
          "environment": "test",
          "pspReference": undefined,
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
